client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import noop from 'lodash/noop';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get, includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Spinner from 'components/spinner';
+import { getCurrentUserEmail } from 'state/current-user/selectors';
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( 'sent' === this.props.emailVerification ||
+			'error' === this.props.emailVerification ) {
+			return this.props.translate( 'Email Sent' );
+		}
+		if ( 'requesting' === this.props.emailVerification ) {
+			return <Spinner className="email-verification-dialog__confirmation-dialog-spinner" />;
+		}
+		return this.props.translate( 'Resend Email' );
+	}
```

Comment:
fun idea! right now every instance of this React component gets a copy of the function in memory but it doesn't really depend on the React instance itself. we could move this out of the React class and into the module scope and pass in the prop. additionally this will make it easier to test

```js
const resendLabel = state => {
	if ( 'sent' === state ) {
		…
	}

	…
}

class VerifyEmailDialog extends Component {
	…
		{ resendLabel( this.props.emailVerification ) }
	…
}
```